frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.tsx

```diff
@@ -176,8 +176,21 @@ export const taxonomicFilterLogic = kea<taxonomicFilterLogicType>({
                     type: TaxonomicFilterGroupType.EventProperties,
                     endpoint: combineUrl(
                         `api/projects/${teamId}/property_definitions`,
-                        featureFlags[FEATURE_FLAGS.UNSEEN_EVENT_PROPERTIES] ? { event_names: eventNames } : {}
+                        featureFlags[FEATURE_FLAGS.UNSEEN_EVENT_PROPERTIES] && eventNames.length > 0
```

Comment:
When I try open the event selector in insight edit mode, the page crashes because of this line: `Uncaught TypeError: Cannot read properties of undefined (reading 'length')`. I guess `eventNames` is an optional prop so in some places it's `undefined`.